programs/merkle-tree/src/transaction_merkle_tree/state.rs

```diff
@@ -1,26 +1,33 @@
 use aligned_sized::aligned_sized;
 use anchor_lang::prelude::*;
-
-use crate::{
-    impl_indexed_merkle_tree,
-    utils::config::{MERKLE_TREE_HEIGHT, MERKLE_TREE_HISTORY_SIZE},
+use light_macros::pubkey;
+use light_merkle_tree::{
+    config::MerkleTreeConfig,
+    constants::{poseidon::ZERO_BYTES, ZeroBytes},
+    hasher::Poseidon,
+    MerkleTree,
 };
 
+use crate::impl_indexed_merkle_tree;
+
+#[derive(Clone, Copy)]
+pub struct TransactionMerkleTreeConfig {}
+
+impl MerkleTreeConfig for TransactionMerkleTreeConfig {
+    const ZERO_BYTES: ZeroBytes = ZERO_BYTES;
+    const PROGRAM_ID: Pubkey = pubkey!("JA5cjkRJ1euVi9xLWsCJVzsRzEkT8vcC4rqw9sVAo5d6");
+}
+
 // NOTE(vadorovsky): This implementation of Merkle tree exists only for
 // transactions and handling Poseidon in multiple rounds. Once Poseidon syscall
 // in Solana is implemented, this implementation will be replaced with
 // light-merkle-tree crate.
 #[account(zero_copy)]
 #[aligned_sized(anchor)]
-#[derive(Eq, PartialEq, Debug)]
 pub struct TransactionMerkleTree {
-    pub filled_subtrees: [[u8; 32]; MERKLE_TREE_HEIGHT],
-    pub current_root_index: u64,
-    pub next_index: u64,
-    pub roots: [[u8; 32]; MERKLE_TREE_HISTORY_SIZE as usize],
+    pub merkle_tree: MerkleTree<Poseidon, TransactionMerkleTreeConfig>,
```

Comment:
Height and root history size is not configurable in our Merkle tree crate?

Reply:
could we add these to TransactionMerkleTreeConfig?
(Height might be unnecessary because of the zero bytes. Not unnecessary if we are using generalized zero bytes of height 32.)

Reply:
a general question how could we support variable Merkle tree sizes?
(Because we will want to switch to a Merkle tree of greater size for sure once poseidon syscall is active.)
Maybe we could always allocate space for a max height of the MerkleTree struct, save the height in a separate variable of the pda and deserialize dynamically depending on the pda tree height.

Reply:
I'm definitely planning to make it configurable, hopefully with a const generic. So far, when I tried that, I was drowning in Anchor errors, so I'm going to try that when I'm unblocked from all the other things and I get green tests.

Reply:
Marking as done - we have a dynamic size now, with a max height of 32.